File: chapter4 - Trees and Graphs/4.3_listOfDepths/Java/dfs/TreeNode.java
package dfs;

public class TreeNode {
  public int val;
  public TreeNode left;
  public TreeNode right;

  public TreeNode(int val) {
    this.val = val;
  }

  public void insert(int val) {
    if (val < this.val) {
      if (this.left == null) {
        this.left = new TreeNode(val);
      } else {
        this.left.insert(val);
      }
    } else {
        if (this.right == null) {
          this.right = new TreeNode(val);
        } else {
          this.right.insert(val);
        }
    }
  }
}

File: chapter4 - Trees and Graphs/4.5_validateBST/JavaScript/TreeNode.js
class TreeNode {
  constructor(val) {
    this.val = val;
    this.left = null;
    this.right = null;
  }

  insert(val) {
    if (val < this.val) {
      if (this.left === null) {
        this.left = new TreeNode(val);
      } else {
        this.left.insert(val);
      }
    } else {
      if (this.right === null) {
        this.right = new TreeNode(val);
      } else {
        this.right.insert(val);
      }
    }
  }
}

module.exports = TreeNode;